### DISCUSSION TITLE
Mirth Connect Hardware Sizing

### DISCUSSION THREAD

**Discussion by dfauraz** [nextgenhealthcare/connect#5480] [Q&A]
Hi!

I know it's a difficult subject (how many channels, how many messages, how much calculation ...)
but is there any resource regarding the right CPU/RAM sizing of a Mirth Connect server ?
We have +/- 400 channels

Thanks !

Daniel

**Comment by jonbartels** [comment]
What is your monitoring and instrumentation telling you? Monitoring tools that show you the JMX statistics (heap usage and thread counts) and basic OS stats (CPU, RAM, IOPS) will be very helpful to allocate resources correctly. At my last job we used Data Dog, my current job uses New Relic and Prometheus, previous customers have used Prometheus and AWS offerings. 

From experience 400 channels would typically have at least 2GB of heap, ideally 4GB. Then 2-4 CPU cores, CPU does not matter as much. Mirth Connect tends to be memory intensive and write intensive.

Mirth Connect is also sensitive to database performance, particularly I/O because it is INSERT and UPDATE heavy.

A subtle issue is that you'll also want to run your *MC Client* with more heap with large channel counts. The client pulls a large XML blob and then has to read it into memory.

**Reply by dfauraz** [reply]
Thanks for your figures, that's a good starting point.
Unfortunately we have Mirth Appliances = no access to vmoptions = no JMX

**Reply by jonbartels** [reply]
Ach! The appliance does make this complicated.

You should have access to alter vmoptions.

Check in the appliance under `folders` then browse to the connect directory. I haven't used an appliance in about a year but I thought there was some access to alter the config files.

Support also has a mechanism to access the appliance as a root user via the SupportNet VPN. They physically can enable other options but I do not know if their internal policies allow it.

The other alternative for monitoring is a horrible hack. You can write a channel that queries the Java MXBeans using code then output that data as JSON. It is far from good monitoring but it can help give you a better idea of your heap usage.

**Reply by dfauraz** [reply]
We will ask support :)

**Reply by aungktun** [reply]
Does Mirth Connect paid version runs on Windows platform ? or only runs on Linux appliance?

**Reply by pacmano1** [reply]
> Does Mirth Connect paid version runs on Windows platform ? or only runs on Linux appliance? 

Please don't hijack threads. The "paid" version includes licensed extensions.  The base software for the paid extensions is the same software available for free that can run on windows or linux.

**Reply by kirbykn2** [reply]
Usually the infrastructure folks have tools(SolarWinds, LogicMonitor, etc) that you use to monitor performance of your servers.  Monitoring trends over time will help you accurately plan for growth.  Additionally, there used to be an expert page for the Mirth appliances(yourserver/expert).  The expert tool has a lot of server resource and troubleshooting tools.
Lastly, you should start the process to migrate off the appliances.  I used the appliances for years, and they worked fine.  But now with containers and things like TeraForm, the appliances are all but obsolete.
